File: module-services/src/main/java/eu/geekhome/services/mqtt/MqttBrokerService.java
package eu.geekhome.services.mqtt;

import org.pf4j.ExtensionPoint;

public interface MqttBrokerService extends ExtensionPoint {
    void addMqttListener(MQTTListener listener);

    void removeMqttListener(MQTTListener listener);

    void publish(String topic, String content);
}

File: module-domain/src/main/java/eu/geekhome/services/mqtt/MqttBrokerService.kt
package eu.geekhome.services.mqtt

import org.pf4j.ExtensionPoint

interface MqttBrokerService : ExtensionPoint {
    fun addMqttListener(listener: MqttListener)
    fun removeMqttListener(listener: MqttListener)
    fun publish(topic: String?, content: String)
}